org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JVMConfigurator.java

```diff
@@ -52,55 +52,45 @@
  */
 public class JVMConfigurator implements IVMInstallChangedListener {
 
-	public static boolean configureDefaultVM(Preferences preferences) throws CoreException {
-		if (preferences == null) {
+	public static boolean configureDefaultVM(String javaHome) throws CoreException {
+		if (StringUtils.isBlank(javaHome)) {
 			return false;
 		}
-		String javaHome = preferences.getJavaHome();
-		boolean changed = false;
-		if (javaHome != null) {
-			File jvmHome = new File(javaHome);
-			if (jvmHome.isDirectory()) {
-				IVMInstall vm = findVM(jvmHome, null);
-				if (vm == null) {
-					IVMInstallType installType = JavaRuntime.getVMInstallType(StandardVMType.ID_STANDARD_VM_TYPE);
-					long unique = System.currentTimeMillis();
-					while (installType.findVMInstall(String.valueOf(unique)) != null) {
-						unique++;
-					}
-					String vmId = String.valueOf(unique);
-					VMStandin vmStandin = new VMStandin(installType, vmId);
-					String name = StringUtils.defaultIfBlank(jvmHome.getName(), "JRE");
-					vmStandin.setName(name);
-					vmStandin.setInstallLocation(jvmHome);
-					vm = vmStandin.convertToRealVM();
-					JDTUtils.setCompatibleVMs(vm.getId());
-					changed = true;
```

Comment:
this caused full builds even though there was already a default runtime environment